MediaBrowser.Common/Providers/ProviderIdParsers.cs

```diff
@@ -0,0 +1,119 @@
+﻿#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace MediaBrowser.Common.Providers
+{
+    /// <summary>
+    /// Parsers for provider ids.
+    /// </summary>
+    public static class ProviderIdParsers
+    {
+        /// <summary>
+        /// Parses an IMDb id from a string.
+        /// </summary>
+        /// <param name="text">The text to parse.</param>
+        /// <param name="imdbId">The parsed IMDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseImdbId(string text, [NotNullWhen(true)] out string? imdbId)
+        {
+            var span = text.AsSpan();
+            var tt = "tt".AsSpan();
+
+            while (true)
+            {
+                var ttPos = span.IndexOf(tt);
+                if (ttPos == -1)
+                {
+                    imdbId = default;
+                    return false;
+                }
+
+                span = span.Slice(ttPos + tt.Length);
+
+                int i = 0;
+                // IMDb id has a maximum of 8 digits
+                int max = span.Length > 8 ? 8 : span.Length;
+                for (; i < max; i++)
+                {
+                    var c = span[i];
+
+                    if (c < '0' || c > '9')
+                    {
+                        break;
+                    }
+                }
+
+                // IMDb id has a minimum of 7 digits
+                if (i >= 7)
```

Comment:
What if the 9th char is also a number. Is it valid then? The regex would've captured it of course...

Reply:
Hmm maybe we shouldn't limit the length. 
https://www.imdb.com/title/tt0004154796/ is the same as https://www.imdb.com/title/tt4154796/. So the only limit is the minimum of 7 digits since https://www.imdb.com/title/tt000001/ is not the same as https://www.imdb.com/title/tt0000001/
Don't know why anyone would add additional zeroes at the beginning but it is possible